### PR TITLE
Custom asset implementation and upgraded trading-pair

### PR DESCRIPTION
**Pull request by ksr30**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -388,34 +386,35 @@ impl<T: Trait> Module<T> {
...
         let orderbook = Orderbook::new(base_asset_id, quote_asset_id, trading_pair_id.clone());
         <Orderbooks<T>>::insert(trading_pair_id, orderbook);
         <AsksLevels<T>>::insert(trading_pair_id, Vec::<FixedU128>::new());
         <BidsLevels<T>>::insert(trading_pair_id, Vec::<FixedU128>::new());
     }
 
     /// Creates a TradingPairID from both Asset IDs.
-    fn create_trading_pair_id(quote_asset_id: &u32, base_asset_id: &u32) -> T::Hash {
+    fn create_trading_pair_id(quote_asset_id: &T::Hash, base_asset_id: &T::Hash) -> T::Hash {
         (quote_asset_id, base_asset_id).using_encoded(<T as frame_system::Trait>::Hashing::hash)
```
Comment (Gauthamastro) [inline src/lib.rs:411]: this is wrong, It should be like if base_id < quote_id then (base_id,quote_id) else (quote_id,base_id). This is how we derive the trading pair

File: src/mock.rs [hunk 2]
```diff
@@ -1,92 +1,92 @@
...
+
+pub type DEXModule = Module<Test>;
+type System = frame_system::Module<Test>;
+
+// Build genesis storage according to the mock runtime.
+pub fn new_test_ext() -> sp_io::TestExternalities {
+    let endowed_accounts: Vec<u64> = vec![1, 2];
+    const UNIT: u128 = 1_000_000_000_000;
+    let mut genesis = system::GenesisConfig::default().build_storage::<Test>().unwrap();
+    pallet_generic_asset::GenesisConfig::<Test> {
```
Comment (Gauthamastro) [inline src/mock.rs:80]: We need to change this to custom_assets.

File: src/tests.rs [hunk 3]
```diff
@@ -1,394 +1,395 @@
...
-fn calculate_quantity(mut pricelevel: LinkedPriceLevel<Test>) -> FixedU128 {
-    let mut total_quantity = FixedU128::from(0);
-    while let Some(order) = pricelevel.orders.pop_back() {
-        total_quantity = total_quantity.checked_add(&order.quantity).unwrap()
-    }
-    total_quantity
-}
\ No newline at end of file
+use codec::Encode;
+use frame_support::{assert_noop, assert_ok};
```
Comment (Gauthamastro) [inline src/tests.rs:2]: We need to convert the test cases to custom_assets too

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (ksr30) [reply src/lib.rs:411]: actually I forgot to remove that ... we are not using that function

Comment (ksr30) [reply src/lib.rs:411]: check https://github.com/Polkadex-Substrate/polkadex-engine/blob/feature/DEX-204.06/src/lib.rs#L527

Comment (Gauthamastro) [reply src/lib.rs:413]: Okay got it.

**Review by Gauthamastro — CHANGES_REQUESTED** [review]
Implement Custom Assets in Mock.rs and Tests.rs

**Comment by ksr30** [comment]
yea .. I will do it after altogether

**Pull request merged**